lib/twilioconnection.js

```diff
@@ -392,6 +404,8 @@ class TwilioConnection extends StateMachine {
    */
   _handleMessage({ body }) {
     if (this.state !== 'open') {
+      this._log.warn(`Unexpected state "${this.state}" for handling a "msg" message`
+        + ' from the TCMP server.');
```

Comment:
You could defer messages while in the connecting state just in case they somehow arrive before the welcome. It shouldn't happen with a WebSocket.

Reply:
@ceaglest This has never happened so far, so I'll defer on this for now.